src/SoundOptionsModal.js

```diff
@@ -0,0 +1,138 @@
+        });
+    };
+
+    handleOnSave = () => {
+        this.props.onChangeSoundOptions(this.state.audioEnabled, this.state.announcementsEnabled, this.state.sonificationEnabled );
+    }
+
+    handleOnCancel = () => {
+        this.setState({
+            audioEnabled: this.props.audioEnabled,
+            announcementsEnabled: this.props.announcementsEnabled,
+            sonificationEnabled: this.props.sonificationEnabled
+        });
+        this.props.onCancel();
+    }
+
+    renderSoundOptions = () => {
+        const soundOptions = [];
+        const soundOptionsProps = [
+            { id:'sound-options-allsounds', label: this.props.intl.formatMessage({ id: 'SoundOptionsModal.allSounds'}), toggleState: this.state.audioEnabled, onChange: this.handleToggleAudioEnabled },
+            { id:'sound-options-musicalSounds', label: this.props.intl.formatMessage({ id: 'SoundOptionsModal.musicalSounds' }), toggleState: this.state.sonificationEnabled, onChange: this.handleToggleSonificationEnabled },
+            { id:'sound-options-announcements', label: this.props.intl.formatMessage({ id: 'SoundOptionsModal.announcements' }), toggleState: this.state.announcementsEnabled, onChange: this.handleToggleAnnouncementsEnabled }
+        ];
+        for (let i=0; i< soundOptionsProps.length; i++) {
+            const soundOptionProps = soundOptionsProps[i];
+            soundOptions.push(
+                <div className='SoundOptions__option-container' key={soundOptionProps.id}>
+                    <div aria-hidden={true}>
+                        {soundOptionProps.label}
+                    </div>
+                    <div className='SoundOptions__toggle-container'>
+                        <div aria-hidden={true}>
+                            {this.props.intl.formatMessage({ id: 'SoundOptionsModal.toggleOff' })}
+                        </div>
+                        <ToggleSwitch
+                            className='SoundOptionsToggle'
+                            id={soundOptionProps.id}
+                            ariaLabel={soundOptionProps.label}
+                            value={soundOptionProps.toggleState}
+                            contentsTrue={null}
+                            contentsFalse={null}
+                            onChange={soundOptionProps.onChange}
+                        />
+                        <div aria-hidden={true}>
+                            {this.props.intl.formatMessage({ id: 'SoundOptionsModal.toggleOn' })}
+                        </div>
+                    </div>
+                </div>
+            )
+        }
+        return soundOptions;
+    }
+
+    render() {
+        return (
+            <ModalWithFooter
+                show={this.props.show}
+                focusOnOpenSelector={'#sound-options-allsounds'}
+                focusOnCloseSelector={'.App__header-soundOptions'}
+                ariaLabel={this.props.intl.formatMessage({ id: 'SoundOptionsModal.title' })}
```

Comment:
I generally assign the formatted string to a variable if I have to use the same key more than once in the same file.

Reply:
I changed ariaLabel to aria-labelledBy, as we have clear target for aria-labelledBy.